script/core/diagnostics/incomplete-signature-doc.lua

```diff
@@ -59,30 +72,34 @@ return function (uri, callback)
 
         local functionName = source.parent[1]
 
-        if source.args and #source.args > 0 then
-            for _, arg in ipairs(source.args) do
-                local argName = arg[1]
-                if argName ~= 'self' then
-                    if not findParam(source.bindDocs, argName) then
-                        callback {
-                            start   = arg.start,
-                            finish  = arg.finish,
-                            message = lang.script('DIAG_INCOMPLETE_SIGNATURE_DOC_PARAM', argName, functionName),
-                        }
+        --- don't apply rule if there is no @param or @return annotation yet
+        --- so comments and @async can be applied without the need for a full documentation
+        if(findSignatureDoc(source.bindDocs)) then
```

Comment:
The indentation level reaches alarming ;-> depth ...
I'd suggest reducing this by directly exiting the function after a check (and, if necessary, extract some more functions for that reason)...

Reply:
Extracted condition, would leave the rest untouched (to minimize changes in the file for the PR)